core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1405,21 +1405,95 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
   // We make it part of each instance rather than the object to facilitate testing.
   private val zkClientConfigViaSystemProperties = new ZKClientConfig()
 
-  override def originals: util.Map[String, AnyRef] =
-    if (this eq currentConfig) super.originals else currentConfig.originals
+  private def maybeMutateOriginalsToMakeBrokerIdAndNodeIdMatch(map: util.Map[String, AnyRef]): util.Map[String, AnyRef] = {
```

Comment:
This is pretty much the same as `maybeMutateOriginalsStringsToMakeBrokerIdAndNodeIdMatch()` below, but I was unable to consolidate them into a single method due to type-related errors.